// @azure/ai-text-analytics/dist-esm/src/generated/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
export const endpoint = {
    parameterPath: "endpoint",
    mapper: {
        required: true,
        serializedName: "Endpoint",
        defaultValue: '',
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const includeStatistics = {
    parameterPath: [
        "options",
        "includeStatistics"
    ],
    mapper: {
        serializedName: "showStats",
        type: {
            name: "Boolean"
        }
    }
};
export const modelVersion = {
    parameterPath: [
        "options",
        "modelVersion"
    ],
    mapper: {
        serializedName: "model-version",
        type: {
            name: "String"
        }
    }
};
